x-pack/platform/plugins/shared/triggers_actions_ui/public/application/sections/rules_list/components/rules_list.tsx

```diff
@@ -452,7 +454,11 @@ export const RulesList = ({
       cloneRuleId.current = null;
       setIsCloningRule(false);
       if (ruleItem) {
-        onRuleEdit(ruleItem);
+        if (navigateToEditRuleForm) {
+          navigateToEditRuleForm(ruleItem.id);
```

Comment:
Same here. Let's move the logic inside the `onRuleEdit` function and keep the code here as before.